### PR TITLE
Add links to contributed custom scalar specs at scalars.graphql.org

### PR DESCRIPTION
**Pull request by dondonz**
The PR follows on from the launch of scalars.graphql.org. 
https://github.com/graphql/graphql-scalars
https://graphql.org/blog/2023-01-14-graphql-scalars/

As discussed in the January APAC working group meeting, this PR adds how to contribute custom scalar specs, links to spec templates, and adds examples.

cc @andimarek

### CODE REVIEW

**Comment by dondonz** [comment]
Feel free to comment on the wording

**Comment by andimarek** [comment]
I am not 100% regarding the process here: I don't see this as the normal trivial editorial change (e.g. typo), but it is also no real change in the spec. Therefore I am leaning to treat it as editorial change. 

Keen to hear what others think @benjie @michaelstaib @leebyron

File: spec/Section 3 -- Type System.md
Comment (benjie) [inline spec/Section 3 -- Type System.md:423]: Let's limit it to just one link; the implementation guide itself should link out to other resources if necessary.

```suggestion
examples in their description; see the GraphQL scalars [implementation guide](https://scalars.graphql.org/implementation-guide) for more guidance.
```

File: spec/Section 3 -- Type System.md
Comment (benjie) [inline spec/Section 3 -- Type System.md:2114]: Let's move this out of the normative section, and reduce it to the one link:

```suggestion
coercion rules. It must not appear on built-in scalar types.

Note: Details on implementing a GraphQL scalar specification can be found in the [scalars.graphql.org implementation guide](https://scalars.graphql.org/implementation-guide).
```

File: spec/Section 3 -- Type System.md
Comment (benjie) [reply spec/Section 3 -- Type System.md:398]: On second thoughts, since we have this detail below, I don't think we need any edits here. Suggest we merge this into the comment on line 415ish below.

**Comment by dondonz** [comment]
Thanks @benjie ! I've put through all your comments

File: spec/Section 3 -- Type System.md
Comment (andimarek) [inline spec/Section 3 -- Type System.md:2123]: I honestly think these are not good examples: we should not encourage people to link directly 
to RFCs. I would even suggest to remove the UUID example. A RFC is great to be the basis of a custom scalar, but it is almost never sufficient to be a good guide. We should people actively encourage to write dedicated custom scalars specs.

Comment (benjie) [reply spec/Section 3 -- Type System.md:2123]: Agreed; writing a scalars.graphql.org spec that calls out to the relevant RFC seems wise. I'm not sure if that needs to be done in this PR though?

Comment (benjie) [reply spec/Section 3 -- Type System.md:2123]: It definitely doesn't need to be addressed in this PR; it's already referenced on line 407, this is just a copy. Can someone file it for follow-up though?

File: spec/Section 3 -- Type System.md
Comment (benjie) [inline spec/Section 3 -- Type System.md:2124]: The text before this block says "In this example, a custom scalar type for UUID is defined with a URL pointing to the relevant IETF specification." therefore the block should only define UUID. The prior examples are sufficient, no need for change here.

```suggestion
```

**Review by benjie — CHANGES_REQUESTED** [review]
Looking good, let's trim those final two lines and then I think we're good to go (after you run `npm install && npm run format` to appease Prettier) 👍

**Comment by dondonz** [comment]
I have spoken the magic words `npm install && npm run format` and I hope Prettier is pleased.

I'm ready for a final review @benjie

**Comment by dondonz** [comment]
Hi @benjie could you please have another review?

**Review by benjie — APPROVED** [review]
Looks good to me; thanks @dondonz! I'd like another approval from @graphql/tsc  and then I think we're good to merge 👍

**Review by mjmahone — APPROVED** [review]
These changes seem clean to me, all in favor of shipping!

**Comment by leebyron** [comment]
Nice work! Glad to have these references in place

**Pull request merged**
